src/checkers/inference/model/serialization/JsonSerializer.java

```diff
@@ -329,6 +336,22 @@ public JSONObject serialize(PreferenceConstraint constraint) {
         return obj;
     }
 
+    @Override
+    public JSONObject serialize(ImplicationConstraint implicationConstraint) {
+        if (implicationConstraint.getAssumptions() == null || implicationConstraint.getConclusion() == null) {
+            return null;
+        }
+        JSONObject obj = new JSONObject();
+        obj.put(CONSTRAINT_KEY, IMPLICATION_CONSTRAINT_KEY);
+        List<JSONObject> assumptions = new ArrayList<>();
```

Comment:
rename variable to serializedAssumptions

Reply:
Done